src/main.js

```diff
@@ -414,10 +418,31 @@ function handleAppWebContentsCreated(dc, contents) {
   });
 
   contents.on('new-window', (event, url) => {
-    if (isTrustedURL(url)) {
+    event.preventDefault();
+    if (!isTrustedURL(url)) {
+      log.info(`Untrusted popup window blocked: ${url}`);
       return;
     }
-    event.preventDefault();
+    if (popupWindow && popupWindow.getURL() === url) {
```

Comment:
Should we log if `popupWindow && popupWindow.getURL() !== url` for debugging purposes? (I'm imagining a popup window that doesn't close for whatever reason.)

Reply:
Done!